### PR TITLE
urjtag: new package

### PR DESCRIPTION
**Pull request by lowfatcomputing**
- Added UrJTAG package
- Added some "use flags" http://blog.lastlog.de/posts/useflags_in_nixos/ (but not yet all possible flags)
- Added myself to maintainers.nix

First package, I'm new to NixOS (coming from ArchLinux).
UrJTAG tends to have very infrequent releases which lag far behind the vcs activity. Is there a way to update to the latest git commit hash without submitting a new merge request here?

### CODE REVIEW

**Comment by bjornfor** [comment]
To do local updates, without copying the whole expression or working in your own nixpkgs fork, you can use ```.overrideDerivation```. See https://nixos.org/wiki/Nix_Modifying_Packages#Overriding_Existing_Packages.

File: pkgs/tools/misc/urjtag/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ pythonBindings ? false
+, jedecSupport ? false
+, python3 ? null
+, stdenv, autoconf, automake, pkgconfig, gettext, intltool, libtool, bison, flex, fetchgit, makeWrapper
```
Comment (bjornfor) [inline pkgs/tools/misc/urjtag/default.nix]: The general style is to have mandatory arguments first, then the optional ones. And wrap at 80 chars. (You picked the right comma-style (comma first on the next line) on first try. Good!)

File: pkgs/tools/misc/urjtag/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+stdenv.mkDerivation rec {
+  name = "urjtag";
+  meta = {
+    description = "Enhanced, modern tool for communicating over JTAG with flash chips, CPUs,and many more";
+    homepage = "http://urjtag.org/";
+    license = with stdenv.lib.licenses; [ gpl2Plus lgpl21Plus ];
+    platforms = stdenv.lib.platforms.gnu;  # arbitrary choice
+    maintainers = with stdenv.lib.maintainers; [ lowfatcomputing ];
+  };
+
```
Comment (bjornfor) [inline pkgs/tools/misc/urjtag/default.nix]: The meta attribute is generally placed at the very end.

File: pkgs/tools/misc/urjtag/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{ pythonBindings ? false
+, jedecSupport ? false
+, python3 ? null
+, stdenv, autoconf, automake, pkgconfig, gettext, intltool, libtool, bison, flex, fetchgit, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  name = "urjtag";
```
Comment (bjornfor) [inline pkgs/tools/misc/urjtag/default.nix]: There should be a version number here; "urjtag-VERSION". If upstream doesn't make releases, use the date (YYYY-MM-DD) of the git commit.

File: pkgs/tools/misc/urjtag/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ${if jedecSupport then "--enable-jedec-exp" else "--disable-jedec-exp"}
+    ${if pythonBindings then "--enable-python" else "--disable-python"}
+  '';
+  preConfigure = "cd urjtag; ./autogen.sh";
+
+  buildPhase = "make";
+
+  checkPhase = "make test";
+
+  installPhase = "make DESTDIR=$out install";
```
Comment (bjornfor) [inline pkgs/tools/misc/urjtag/default.nix]: I think the default buildPhase and checkPhase do exactly what you do (so they can be removed). installPhase can alternatively be fixed by adding ```makeFlags = [ "DESTDIR=$(out)" ]```, but I guess either way is OK.

File: pkgs/tools/misc/urjtag/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    url = "git://git.code.sf.net/p/urjtag/git";
+    rev = "7ba12da7845af7601e014a2a107670edc5d6997d";
+    sha256 = "834401d851728c48f1c055d24dc83b6173c701bf352d3a964ec7ff1aff3abf6a";
+  };
+
+  buildInputs = [ gettext pkgconfig autoconf automake libtool makeWrapper ]
+    ++ stdenv.lib.optional pythonBindings python3;
+
+  configureFlags = ''
+    --prefix=/
```
Comment (bjornfor) [inline pkgs/tools/misc/urjtag/default.nix]: Out of curiosity, did you try flipping "--prefix=/" and "DESTDIR=$out" with "--prefix=$out" and removing DESTDIR? I think that's how it's most often done, but of course upstream build systems vary :-)

File: pkgs/tools/misc/urjtag/default.nix (hunk 3 above)
Comment (lowfatcomputing) [inline pkgs/tools/misc/urjtag/default.nix]: @bjornfor The UrJTAG project does make releases. [Looks like the latest release (0.10) was on April 17th](http://urjtag.org/). Maybe I should just use 0.10 for the version string? It sounds like sha256 hashes are more important and metadata version strings are purely informational (for humans) anyway...?

File: pkgs/tools/misc/urjtag/default.nix (hunk 5 above)
Comment (lowfatcomputing) [inline pkgs/tools/misc/urjtag/default.nix]: > Out of curiosity, did you try flipping "--prefix=/" and "DESTDIR=$out" with "--prefix=$out" and removing DESTDIR?

Just tried that but it doesn't seem to work. So I guess I should go with this method: http://github.com/NixOS/nixpkgs/pull/9533#discussion_r38264557

**Comment by lowfatcomputing** [comment]
Uh-Oh looks like someone else's commits are included in this pull request. Anyone know how to fix this?

**Comment by dfoxfranke** [comment]
@lowfatcomputing Rebase your branch against master.

**Comment by bjornfor** [comment]
@lowfatcomputing: Yes, 0.10 as version string is correct then. Please squash the commits into one "urjtag: init at 0.10". Yes, we have a commit message style guide :-)

**Comment by lowfatcomputing** [comment]
@dfoxfranke: Alright I think we're good now, thanks. Sorry about the git-goof(!) @bjornfor, thanks for your nix-tips. Anything pending?

**Comment by bjornfor** [comment]
Looks good. Pushed to master (e7a024abc4ff928ab56b059a36eba7da98579ad9). Thanks.

**Pull request closed without merging**